libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.kt

```diff
@@ -660,9 +667,16 @@ open class OAuthWebviewHelper : KeyChainAliasCallback {
                         null
                     )
 
-                    else -> when {
-                        instance.useWebServerAuthentication -> onWebServerFlowComplete(params["code"])
-                        else -> onAuthFlowComplete(TokenEndpointResponse(params))
+                    else -> {
+                        // Determine if presence of override parameters require the user agent flow.
+                        val overrideWithUserAgentFlow = isUsingFrontDoorBridgeUrl && frontDoorBridgeCodeVerifier == null
+                        when {
+                            instance.useWebServerAuthentication && !overrideWithUserAgentFlow ->
```

Comment:
💤 [No after hours reply needed] @brandonpage & @wmathurin, this is my current thinking for storing the front door bridge URL and its optional accompanying code verifier as dedicated properties so we may then use them to determine the authentication flow between web server and user agent.  That's for this thread from earlier:

https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2594#discussion_r1688807301

This looks a lot like what I have in place for iOS in SFOAuthCoordinator.

Just having an outside force overriding the app's chosen auth flow is complicated, so I want to internalize that from the app and try to keep our code simple.  I was hoping just storing those and then clearly determining this one local variable would keep that readable.  I'm open the suggestions, though.

On the iOS side, SFOAuthCoordinator is the WKNavigationDelegate.  It would possibly be a lot easier to maintain if that was a separate class chosen just when a front door URL is in place.  I'm thinking "one class per job" rather than a single class with an overly complicated control flow.  I _didn't_ do that yet, but perhaps when we convert that to Swift and here we eliminate OAuthWebViewHelper we could really simplify the code for this complicated set of logic flow.